### PR TITLE
Ship with Empty Starter

### PR DESCRIPTION
**Pull request by freddyucv**
In 5.3.0, we need to ship dotCMS with the empty starter by default.

Some test was fix because these was using data from the starter:

- The Publish Action ID change in this new starter so, I have to change this test t use the right value:

https://github.com/dotCMS/core/pull/18337/commits/ce1b8496447efe9b1e315ad644e159c31bf9bbf9

- The Save & Publish Action was removed in the new starter

https://github.com/dotCMS/core/pull/18337/commits/614e58a0d39410ada401064d090a458ea0044526

Some Fix was made in SaveContentAsDraftActionlet:

- When a Contentlet with categories was updated sending a null categories list,  it lost all the categories
Expected: the COntentlet should keep the categories, to delete all the categories from a contentlet you should sent a empty list

https://github.com/dotCMS/core/pull/18337/commits/46986e7b5b3f2d3203607eda9f0fde6b7e45699c

-  Tags Field value not Updating when saving as draft

https://github.com/dotCMS/core/pull/18337/commits/9d4173b9eabf075e95ca768813d98ce9267e1329

- When a new contentlet with Relationship was saved it lost the Relationship field values

https://github.com/dotCMS/core/pull/18337/commits/a764524e3e3fe380eafb97c2f671741337a74618

### CODE REVIEW

**Review by jdotcms — COMMENTED** [review]
Missing comment on the description

**Review by erickgonzalez — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java [hunk 1]
```diff
@@ -72,11 +75,22 @@ private TestCase(
 
     @DataProvider
     public static Object[] usersAndContentTypeWithoutHostField() throws Exception {
+        IntegrationTestInitService.getInstance().init();
+
+        final Optional<WorkflowAction> optionalSaveAction = getWorkflowActionByName("Save");
+        final Optional<WorkflowAction> optionalPublishAction = getWorkflowActionByName("Publish");;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java:81]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java:81]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java:81]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java:81]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java:81]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java:81]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java [hunk 2]
```diff
@@ -116,6 +117,7 @@ public void executeAction(final WorkflowProcessor processor,
 
 			if (categories == null) {
 				categories = this.categoryAPI.getParents(contentlet, user, respectFrontendPermission);
+				categories = UtilMethods.isSet(categories) ? categories : null;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java:120]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java:81]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java:120]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java:120]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java:81]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java [hunk 3]
```diff
@@ -110,12 +111,11 @@ public void executeAction(final WorkflowProcessor processor,
...
 
 			//Keeps existing relationships
-			if (contentletRelationships == null){
-                contentletRelationships = contentletAPI.getAllRelationships(contentlet);
-            }
+
 
 			if (categories == null) {
 				categories = this.categoryAPI.getParents(contentlet, user, respectFrontendPermission);
+				categories = UtilMethods.isSet(categories) ? categories : null;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java:118]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java [hunk 4]
```diff
@@ -110,12 +111,11 @@ public void executeAction(final WorkflowProcessor processor,
 					contentletDependencies.isRespectAnonymousPermissions() : user.isFrontendUser();
 
 			//Keeps existing relationships
-			if (contentletRelationships == null){
-                contentletRelationships = contentletAPI.getAllRelationships(contentlet);
-            }
+
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java:114]: With this change, the Save as Draft functionality still works?

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java:118]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java:81]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java:118]: Codacy Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java:81]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)

**Review by jdotcms — APPROVED** [review]

**Pull request merged**